tools/importer-rest-api-specs/components/dataapigeneratorjson/templater_operations.go

```diff
@@ -78,10 +50,10 @@ func codeForOperation(operationName string, operation models.OperationDetails, r
 		resourceId = *operation.ResourceIdName
 	}
 
-	responseObject := ObjectDefinition{}
+	responseObject := operationModels.ObjectDefinition{}
 	if operation.ResponseObject != nil {
 		responseObject.ReferenceName = operation.ResponseObject.ReferenceName
-		responseObject.Type = ObjectDefinitionType(operation.ResponseObject.Type)
+		responseObject.Type = operationModels.ObjectDefinitionType(operation.ResponseObject.Type)
 
 		// TODO
```

Comment:
Theoretically these are different things but in practice from what I've seen we do the same with the result. Perhaps instead it'd be better to indicate whether the response is a single object or an array of objects? (Noting that paged and an array are not necessarily the same - not all endpoints may support paging)

Reply:
Both Requests and Responses can be infinitely scoped (in theory) e.g. a List of a List of a Dictionary of string:float values - but in addition, Request and Response are both optional - so we need to both handle both recursively - and handle them being nilable/not having a default value

Example: some DELETE operations have neither a request or a response model - they're simply a DELETE returning an LRO or not